api/v1alpha7/openstackmachine_types.go

```diff
@@ -107,6 +113,12 @@ type OpenStackMachineStatus struct {
 	// Addresses contains the OpenStack instance associated addresses.
 	Addresses []corev1.NodeAddress `json:"addresses,omitempty"`
 
+	// ImageID contains the ID of the image this machine was created with.
+	ImageID string `json:"imageID,omitempty"`
```

Comment:
Why are we also adding `ImageID` here? It seems like a good addition, but does it need to happen in this PR?